lib/SPIRV/SPIRVLowerSaddWithOverflow.h

```diff
@@ -35,22 +35,21 @@
 #ifndef SPIRV_SPIRVLOWERSADDWITHOVERFLOW_H
 #define SPIRV_SPIRVLOWERSADDWITHOVERFLOW_H
 
-#include "llvm/IR/InstVisitor.h"
 #include "llvm/IR/PassManager.h"
 #include "llvm/Pass.h"
 
 namespace SPIRV {
 
-class SPIRVLowerSaddWithOverflowBase
-    : public llvm::InstVisitor<SPIRVLowerSaddWithOverflowBase> {
+class SPIRVLowerSaddWithOverflowBase {
 public:
   SPIRVLowerSaddWithOverflowBase() : Context(nullptr), Mod(nullptr) {}
-  virtual ~SPIRVLowerSaddWithOverflowBase() {}
-  virtual void visitIntrinsicInst(llvm::CallInst &I);
 
   bool runLowerSaddWithOverflow(llvm::Module &M);
 
 private:
+  void replaceSaddOverflow(llvm::Function &F);
```

Comment:
THinking if lowerSaddOverflow/lowerSaddSat will be better names?

Reply:
Were the parameter an `llvm::IntrinsicInst`, I would agree with you, but the function declarations are replacing the function definition, so I think "replace" is a better verb here than "lower."